Add a limit to what a player can earn over a time limit

hertzigger opened the pull request:
JobsDriver complains that it didn't implement method getParentLogger()

remove comment

limits now working on a global basis, will stop players from earning more
once they get to a defined amount using max-daily-wage in
generalConfig.yml will not limit if max-daily-wage is set < 1

added pay period so limit cna be set over a user defined range and hooked
into the action class to do the payment

update build number

zford commented:
Check Travis, your commit doesn't build.

hertzigger commented:
I am doing, not sure why that is doing it.When I first pulled JobsDriver complained it didn't implement method getParentLogger(). Also I've got a bug that i'm trying to sort out that is quite serious

hertzigger commented:
Hi,

I fairly new to git and the github concept, It looks like i have squashed my commits too far back or something, not sure if that will be a problem. Also i can't reproduce the compile error that Travis is having. I had the same error when i initially pulled. But as you can see I fixed that at JobsDriver complains that it didn't implement method getParentLogger().

C:\Users\Jonathan\Documents\GitHub\Jobs>mvn clean install
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for
me.zford:jobs:jar:2.12.4-SNAPSHOT
[WARNING] 'dependencies.dependency.systemPath' for net.milkbowl.vault:Vault:jar
should not point at files within the project directory, ${basedir}/lib/Vault-1.2
.12.jar will be unresolvable by dependent projects @ line 59, column 19
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten t
he stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support buildin
g such malformed projects.
[WARNING]
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Jobs 2.12.4-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ jobs ---
[INFO] Deleting C:\Users\Jonathan\Documents\GitHub\Jobs\target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ jobs ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to .
[INFO] Copying 12 resources to .
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:compile (default-compile) @ jobs ---
[INFO] Compiling 70 source files to C:\Users\Jonathan\Documents\GitHub\Jobs\targ
et\classes
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ jo
bs ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\Jonathan\Documents\GitHub\Jo
bs\src\test\resources
[INFO]

## [INFO] --- maven-compiler-plugin:2.3.2:testCompile (default-testCompile) @ jobs

[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ jobs ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ jobs ---
[INFO] Building jar: C:\Users\Jonathan\Documents\GitHub\Jobs\target\Jobs.jar
[INFO]
[INFO] --- maven-license-plugin:1.10.b1:check (default) @ jobs ---
[INFO] Checking licenses...
[INFO]
[INFO] --- maven-install-plugin:2.4:install (default-install) @ jobs ---
[INFO] Installing C:\Users\Jonathan\Documents\GitHub\Jobs\target\Jobs.jar to C:\
Users\Jonathan.m2\repository\me\zford\jobs\2.12.4-SNAPSHOT\jobs-2.12.4-SNAPSHOT
.jar
[INFO] Installing C:\Users\Jonathan\Documents\GitHub\Jobs\pom.xml to C:\Users\Jo
nathan.m2\repository\me\zford\jobs\2.12.4-SNAPSHOT\jobs-2.12.4-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.417s
[INFO] Finished at: Sat Aug 24 17:05:06 BST 2013
[INFO] Final Memory: 14M/183M
[INFO] ------------------------------------------------------------------------

zford commented:
You're probably building against Java 7.  We build against Java 6 (look at the pom.xml file)

hertzigger commented:
fixed :)

zford commented:
Your values in JobsPlayer aren't persistant, which means you can bypass the limit several ways

1) Server restart will reset limit.
2) Logging out, then back in if "save-on-disconnect" is set to true.
3) Logging out, waiting a few minutes (no more than "save-period" minutes), then logging back in.

This is everything off the top of my head by looking at it for 30 seconds without testing, there may be more, I'd go back and look for more use cases and test.

Please clean up and squash your commits, rebase on top of the latest.  You're based off an old commit, then merged back into mainline, then back from and old commit back to yourself?  It's weird.  Rebase on top of mainline and squash.

hertzigger committed:
Bit of a mess but simply adding a limit on to how much a user can earn

Squashed commit of the following:

commit 025b9a26d616d4270c3288d143794d7be3f2a1ac
Author: hertzigger <tornadojon@gmail.com>
Date:   Sat Aug 24 18:42:57 2013 +0100

    changed maven java version to 6 and removed overridden method

commit c436c98fbd12cb9ab2a7c224ac7a06f5ac82a06a
Merge: 93ec670 5ba1cc8
Author: hertzigger <tornadojon@gmail.com>
Date:   Sat Aug 24 16:53:09 2013 +0100

    Merge branch 'master' of https://github.com/hertzigger/Jobs

commit 93ec670b4d77e1cb480a40d48134851315c41780
Author: Zak Ford <zak.j.ford@gmail.com>
Date:   Sat Jul 27 08:50:40 2013 -0400

    Use event.getEnchanter()

    update .gitignore for intellij

    JobsDriver complains that it didn't implement method getParentLogger()

    remove comment

    limits now working on a global basis, will stop players from earning more
    once they get to a defined amount using max-daily-wage in
    generalConfig.yml will not limit if max-daily-wage is set < 1

    added pay period so limit cna be set over a user defined range and hooked
    into the action class to do the payment

    update build number

    fixed NullPointerException that was throw when a uses didn't have a job

commit 5ba1cc8c6a7f149518c5ba5747be9be9aaa9688b
Merge: 6a1f6c4 79d4326
Author: hertzigger <tornadojon@gmail.com>
Date:   Sat Aug 24 16:43:06 2013 +0100

    Merge branch 'master' of https://github.com/hertzigger/Jobs

    Conflicts:
    	src/main/java/me/zford/jobs/Jobs.java

commit 6a1f6c4d7c6982135a0b5604e3f37df83548be54
Author: hertzigger <tornadojon@gmail.com>
Date:   Sun Aug 18 20:15:19 2013 +0100

    update .gitignore for intellij

    JobsDriver complains that it didn't implement method getParentLogger()

    remove comment

    limits now working on a global basis, will stop players from earning more
    once they get to a defined amount using max-daily-wage in
    generalConfig.yml will not limit if max-daily-wage is set < 1

    added pay period so limit cna be set over a user defined range and hooked
    into the action class to do the payment

    update build number

    fixed NullPointerException that was throw when a uses didn't have a job

commit 79d4326a258dae04745c02b3dd4557df6b684b64
Author: hertzigger <tornadojon@gmail.com>
Date:   Sun Aug 18 20:15:19 2013 +0100

    update .gitignore for intellij

    JobsDriver complains that it didn't implement method getParentLogger()

    remove comment

    limits now working on a global basis, will stop players from earning more
    once they get to a defined amount using max-daily-wage in
    generalConfig.yml will not limit if max-daily-wage is set < 1

    added pay period so limit cna be set over a user defined range and hooked
    into the action class to do the payment

    update build number

hertzigger commented:
i'm fixing these additional issues

zford commented:
Same issues still persist.  I'm not comfortable with this change without persistance.  Currently this patch doesn't even have logout persistance, much less restart persistence.
